src/backend/services/docker_service.py

```
@@ -71,17 +78,62 @@ def get_docker_client():
+            
+            try:
+                pull_logs = client.api.pull(REMOTE_IMAGE, stream=True, decode=True)
+                for log in pull_logs:
+                    if 'status' in log:
+                        status_msg = log['status']
+                        if 'progress' in log:
+                            status_msg += f" {log['progress']}"
+                        logging.info(f"🐳 Pull: {status_msg}")
+                        yield {"status": "running", "log": status_msg}
+                    if 'error' in log:
+                        logging.error(f"Docker pull error: {log['error']}")
```

Ruff (0.14.4)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)